src/Admin/AdminInterface.php

```diff
@@ -225,10 +226,17 @@ public function getIdParameter();
      */
     public function hasRoute($name);
 
+    //    NEXT_MAJOR: declare native types in 4.0
+
+    /**
+     * @param AclSecurityHandlerInterface $securityHandler
+     *
+     * @return void
+     */
     public function setSecurityHandler(SecurityHandlerInterface $securityHandler);
 
     /**
-     * @return SecurityHandlerInterface|null
+     * @return AclSecurityHandlerInterface|null
```

Comment:
because this one occures
```text
 ------ ------------------------------------------------------------------------------------------------------------------- 
  Line   Util/AdminObjectAclData.php                                                                                        
 ------ ------------------------------------------------------------------------------------------------------------------- 
  263    Call to an undefined method Sonata\AdminBundle\Security\Handler\SecurityHandlerInterface::getObjectPermissions().  
 ------ ------------------------------------------------------------------------------------------------------------------- 
```


fixing it in `AdminObjectAclData` with `assert`

Reply:
You can fix it by following https://github.com/sonata-project/SonataAdminBundle/pull/6161#discussion_r443222626

```
public function __construct(
        AdminInterface $admin,
        $object,
        \Traversable $aclUsers,
        $maskBuilderClass,
        ?\Traversable $aclRoles = null
    ) {
        if (!$admin-> isAclEnabled()) {
            // Throw something
        }

        ...
    }
```

Then 
```
/**
     * Gets security handler.
     *
     * @return AclSecurityHandlerInterface
     */
    public function getSecurityHandler()
```

And then, use `$this->getSecurityHandler` instead of `$this->admin->getSecurityHandler`

Reply:
I've asserted all the things